### ISSUE TITLE
DualPivotFinderTester fails for multiple possible pivots and undefined behaviour

### ISSUE THREAD

**Issue by N0W0RK** [N0W0RK/TUM_GAD_Tests_SS21#44]
**Is your test request linked to a subexercise? Please describe:**
 This is an issue with the dual MedianDistributed Pivot Tests

**Describe the behaviour you want:**
* Multiple possible pivots:
When a value appears multiple times in the considered elements, the tests only allow for one of those elements to be returned as pivots.

    In [this case](https://github.com/N0W0RK/TUM_GAD_Tests_SS21/blob/fd556c8e3ca1f2d898f563dece16ed53ca7e8994/GAD%206%20-%20Verbessertes%20Sortieren/tests/simplesort/DualPivotFinderTester.java#L1856) for example:
    ```
    Considered Elements: [ -9, -28, 13, -28, -22]
    At indices           [  9,  14, 19,  24,  29]
    ```
    The pivots would be -28 and  -9.
    The Test only accepts 24 as index for -28 but 14 would in my opinion also be correct.

* Undefined behaviour
According to artemis, results for arrays smaller then 5 only have to make sense. This implies that the indices have to be inside the given range. [The tests](https://github.com/N0W0RK/TUM_GAD_Tests_SS21/blob/fd556c8e3ca1f2d898f563dece16ed53ca7e8994/GAD%206%20-%20Verbessertes%20Sortieren/tests/simplesort/DualPivotFinderTester.java#L1770) however only permit certain values for range sizes smaller then 5 and notr just any values inside the range.

**Comment by N0W0RK** [comment]
This also applies to:
 * DualPivotFinderTester.getMedianPivotFront(): [1](https://github.com/N0W0RK/TUM_GAD_Tests_SS21/blob/fd556c8e3ca1f2d898f563dece16ed53ca7e8994/GAD%206%20-%20Verbessertes%20Sortieren/tests/simplesort/DualPivotFinderTester.java#L1021)/[2](https://github.com/N0W0RK/TUM_GAD_Tests_SS21/blob/fd556c8e3ca1f2d898f563dece16ed53ca7e8994/GAD%206%20-%20Verbessertes%20Sortieren/tests/simplesort/DualPivotFinderTester.java#L869)

 * DualPivotFinderTester.medianPivotFront(): [1](https://github.com/N0W0RK/TUM_GAD_Tests_SS21/blob/fd556c8e3ca1f2d898f563dece16ed53ca7e8994/GAD%206%20-%20Verbessertes%20Sortieren/tests/simplesort/DualPivotFinderTester.java#L275)/[2](https://github.com/N0W0RK/TUM_GAD_Tests_SS21/blob/fd556c8e3ca1f2d898f563dece16ed53ca7e8994/GAD%206%20-%20Verbessertes%20Sortieren/tests/simplesort/DualPivotFinderTester.java#L345)

**Comment by N0W0RK** [comment]
Probably also medianPivotDistributed() havent tested tho.

**Comment by babyygemperor** [comment]
> * Multiple possible pivots:
When a value appears multiple times in the considered elements, the tests only allow for one of those elements to be returned as pivots.

In this case for example:

Considered Elements: [ -9, -28, 13, -28, -22]
At indices           [  9,  14, 19,  24,  29]
The pivots would be -28 and -9.
The Test only accepts 24 as index for -28 but 14 would in my opinion also be correct.

You do make a good point, my tests are very strict in only accepting when the 2nd smallest element gets returned. Since technically -28 at index 24 is the second element, that's why my tests were expecting it, however I've added the possibility of both the variants to be expected.

> * Undefined behaviour
According to artemis, results for arrays smaller then 5 only have to make sense. This implies that the indices have to be inside the given range. The tests however only permit certain values for range sizes smaller then 5 and notr just any values inside the range.

Well, my tests definitely make sense because they expect the 2nd smallest and 2nd largest element to be returned (even with sizes less than 5), and since I have 100% on Artemis. I'm quite sure Artemis accepts multiple implementations of a solution. With the implementation when you return the 2nd smallest and 2nd largest element (which strictly per definition) makes sense and is therefore considered as correct. The tests from @mertcorumlu expect a different valid implementation. Therefore I believe that both the implementations should be left as an option to the student to decide which one s/he wants to use. The docs above the tests concerning it warns the student about the expected values as well.

However for that particular test, I've added both the variants as excepted answers


> This also applies to:
>   * DualPivotFinderTester.getMedianPivotFront(): 1/2
Same reasons, fixed however anyways



> This also applied to:
>    * DualPivotFinderTester.medianPivotFront(): 1/2

Same reasons, fixed however anyways
